packages/node-cli/src/commands/ManageAccountCommandGenerator.js

```diff
@@ -0,0 +1,271 @@
+            showValidationResults(
+               fieldValue,
+               validateFieldIsNotEmpty,
+               validateFieldHasNoSpaces,
+               (fieldValue) => validateSameAuthID(fieldValue, originalAuthId),
+               (fieldValue) => validateAuthIDNotInList(fieldValue, Object.keys(authIDList)),
+               validateAlphanumericHyphenUnderscore,
+               validateMaximumLength
+            ),
+      });
+      return answer[ANSWERS_NAMES.RENAMETO];
+   }
+
+   async _confirmRemove(prompt) {
+      let answer = await prompt([
+         {
+            type: CommandUtils.INQUIRER_TYPES.LIST,
+            name: ANSWERS_NAMES.REMOVE,
+            message: NodeTranslationService.getMessage(QUESTIONS.VERIFY_REMOVE),
+            default: false,
+            choices: [
+               { name: NodeTranslationService.getMessage(YES), value: true },
+               { name: NodeTranslationService.getMessage(NO), value: false },
+            ],
+         },
+      ]);
+      if (!answer[ANSWERS_NAMES.REMOVE]) {
+         throw NodeTranslationService.getMessage(MESSAGES.CANCEL);
+      }
+      return answer[ANSWERS_NAMES.REMOVE];
+   }
+
+   _extractAnswers(answers) {
+      var commandAnswers = new Object();
+      if (answers[ANSWERS_NAMES.ACTION] == ACTION.RENAME) {
+         commandAnswers[COMMAND.OPTIONS.RENAME] = answers[ANSWERS_NAMES.SELECTED_AUTH_ID].authId;
+         commandAnswers[COMMAND.OPTIONS.RENAMETO] = answers[ANSWERS_NAMES.RENAMETO];
+      } else if (answers[ANSWERS_NAMES.ACTION] == ACTION.REMOVE) {
+         commandAnswers[COMMAND.OPTIONS.REMOVE] = answers[ANSWERS_NAMES.SELECTED_AUTH_ID].authId;
+         // } else if (answers[ANSWERS_NAMES.ACTION] == ACTION.REVOKE) {
+         //    commandAnswers[COMMAND.OPTIONS.REVOKE] = answers[ANSWERS_NAMES.SELECTED_AUTH_ID].authId;
+      }
+      return commandAnswers;
+   }
+
+   async _executeAction(answers) {
+      const sdkParams = CommandUtils.extractCommandOptions(answers, this._commandMetadata);
+      const flags = [];
+      const executionContext = new SdkExecutionContext({
+         command: this._commandMetadata.sdkCommand,
+         params: sdkParams,
+         flags,
+      });
+
+      let message = this._getSpinnerMessage(answers);
+
+      const operationResult = await executeWithSpinner({
+         action: this._sdkExecutor.execute(executionContext),
+         message: message,
+      });
+
+      return operationResult.status === SdkOperationResultUtils.STATUS.SUCCESS
+         ? ManageAccountActionResult.Builder.withData(this.prepareData(answers, operationResult.data))
+              .withResultMessage(operationResult.resultMessage)
+              .build()
+         : ManageAccountActionResult.Builder.withErrors(
+              SdkOperationResultUtils.collectErrorMessages(operationResult)
+           ).build();
+   }
+
+   _getSpinnerMessage(answers) {
+      let message = "";
+      if (answers[COMMAND.OPTIONS.REMOVE]) {
```

Comment:
a `switch` statement would be better for this, since we have 4 (and 5 in the future, with REVOKE) options

Reply:
Use of switch is considered not good practice in terms of clean code. I understand that using if else is basically the same and the solution would be to implement polymorfism but I think this it's not worth enough for overcomplicating

Anyways, condition is that the object "answers" has a property. Answers could have several properties at the same time (like in the case of rename and renameto) so I don't think it would be possible to do this with the switch or it would increase more the complexity.